projects/core/src/lib/map/map-layer.component.ts

```diff
@@ -309,25 +321,37 @@ export class OlRasterLayerComponent
             wrapX: false,
         });
 
-        this.source.setTileLoadFunction((tile, src) => {
+        const proj = olGetProj(this.spatialReference.srsString) as Projection;
+        const tileGrid = this.source.getTileGridForProjection(proj);
+
+        this.source.setTileLoadFunction((olTile, src) => {
+            const tile = olTile as OlImageTile;
+            const tileCoord = tile.getTileCoord();
+            const tileZoomLevel = tileCoord[0];
+            const tileExtent = tileGrid.getTileCoordExtent(tileCoord) as Extent;
+
+            const tileResolution = this.mapService.getView().getResolutionForZoom(tileZoomLevel);
```

Comment:
You don't need this here, since you only use this as an input to `createQueryAbortStream`.

But the `ProjectService` already has this as a dependency, so you can take this from there.

This also means you can revert adding the map service to the layer.

Reply:
4c39c8f4c743670e5368119dd47061314c52246d